deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/layers/LossLayer.java

```diff
@@ -90,6 +93,12 @@ public ParamInitializer initializer() {
         return EmptyParamInitializer.getInstance();
     }
 
+    @Override
+    public @NonNull SDVariable defineLayer(@NonNull SameDiff sameDiff, @NonNull SDVariable layerInput,
+            @NonNull Map<String, SDVariable> paramTable, SDVariable mask) {
+        return doActivation(layerInput);
```

Comment:
You're thinking define the losses in the layers, rather than in the toSameDiff method?  The only issue I see there is that there's no way for the layer to be aware of `minibatch`/`average`.